app/validators/secure_context_uri_validator.rb

```diff
@@ -25,20 +25,43 @@
 #
 # See COPYRIGHT and LICENSE files for more details.
 #++
-class NextcloudSecureHostValidator < ActiveModel::EachValidator
-  # A "secure" Nextcloud host is either a host via https or
-  # localhost/127.0.0.1. Ports are ignored.
+
+# Please see https://w3c.github.io/webappsec-secure-contexts/
+# for a definition of "secure contexts".
+# Basically, a host has to have either a HTTPS scheme or be
+# localhost to be secure.
+class SecureContextUriValidator < ActiveModel::EachValidator
   def validate_each(contract, attribute, value)
     begin
       uri = URI.parse(value)
     rescue StandardError
+      uri = nil
+    end
+
+    if uri.nil? || uri.host.nil?
```

Comment:
`uri&.host.present?` would do the trick. Not sure if that would improve readability by much.

Reply:
I hope my latest changes make it easier to read.